source/gameengine/GameLogic/SCA_VibrationActuator.cpp

```diff
@@ -62,16 +64,32 @@ bool SCA_VibrationActuator::Update()
 		return false;
 	}
 
-	bool bPositiveEvent = IsPositiveEvent();
+	if (IsPositiveEvent()) {
+		switch (m_mode) {
+			case KX_ACT_VIBRATION_PLAY:
+			{
+				instance->RumblePlay(m_strength, m_strength_right, m_duration);
+				m_endtime = PIL_check_seconds_timer() * 1000.0f + m_duration;
+				break;
+			}
+			case KX_ACT_VIBRATION_STOP:
+			{
+				instance->RumbleStop();
+				m_endtime = 0.0f;
+				break;
+			}
+		}
 
-	if (bPositiveEvent) {
-		instance->RumblePlay(m_strength, m_duration);
-		m_endtime = PIL_check_seconds_timer() * 1000.0f + m_duration;
 	}
 
 	RemoveAllEvents();
 
-	return PIL_check_seconds_timer() * 1000.0f < m_endtime;
+	if (!(PIL_check_seconds_timer() * 1000.0f < m_endtime)) {
+		instance->RumbleStop();
+		m_endtime = 0.0f;
+	}
+
+	return instance->GetRumbleStatus();
```

Comment:
I don't know why m_endtime switch from 0.0f to pil time + m_duration each frame in my test file...